src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/HttpContentJsonExtensions.cs

```diff
@@ -37,21 +39,61 @@ public static partial class HttpContentJsonExtensions
 
         private static async Task<object?> ReadFromJsonAsyncCore(HttpContent content, Type type, Encoding? sourceEncoding, JsonSerializerOptions? options, CancellationToken cancellationToken)
         {
-            Stream contentStream = await ReadHttpContentStreamAsync(content, cancellationToken).ConfigureAwait(false);
+            using (Stream contentStream = await GetContentStream(content, sourceEncoding, cancellationToken).ConfigureAwait(false))
+            {
+                return await JsonSerializer.DeserializeAsync(contentStream, type, options ?? JsonContent.s_defaultSerializerOptions, cancellationToken).ConfigureAwait(false);
```

Comment:
This is related to using the new `DeserializeAsync()` and not the code-gen effort, correct?

Reply:
Hm no, it isn't related to the new `DeserializeAsyncEnumerable` method. This diff is just refactoring the code to use a new `GetContentStream` method which contains logic that was previously duplicated, and would have been duplicated more in the new methods I'm adding. This particular method retains its call to the existing `DeserializeAsync`.